relentless/potential/pair.py

```diff
@@ -561,3 +561,244 @@ def _derivative(self, param, r, epsilon, kappa, **params):
+        elif param == 'sigma_i':
+            p1 = sigma_d**2*(sigma_i**2 + 2*sigma_i*sigma_j + 7*sigma_j**2)/(12*(sigma_i + sigma_j)**2)
+            p2 = ((-sigma_d*sigma_j*sigma_i**3 - 3*sigma_d*sigma_i**2*sigma_j**2
+                   - 3*sigma_d*sigma_i*sigma_j**3 - 2*sigma_d*sigma_i*sigma_j**2*r[np.less(r,lo)]
+                   - sigma_d*sigma_j**4 + 2*sigma_d*sigma_j**3*r[np.less(r,lo)] - sigma_d**2*sigma_j**3
+                   + 2*sigma_d**2*sigma_j**2*r[np.less(r,lo)])
+                 /(8*(sigma_i + sigma_j)**3))
+            d[np.less(r,lo)] = np.pi*P*(p2 - p1)
+        elif param == 'sigma_j':
+            p1 = sigma_d**2*(sigma_j**2 + 2*sigma_i*sigma_j + 7*sigma_i**2)/(12*(sigma_i + sigma_j)**2)
+            p2 = ((-sigma_i*sigma_d*sigma_j**3 - 3*sigma_d*sigma_i**2*sigma_j**2 - sigma_j*sigma_i**2*sigma_d**2
+                   - 3*sigma_j*sigma_d*sigma_i**3 + 2*sigma_i**2*sigma_j*sigma_d*r[np.less(r,lo)]
+                   - 2*sigma_i*sigma_j*sigma_d**2*r[np.less(r,lo)] - sigma_d**2*sigma_i**3 - sigma_d*sigma_i**4
+                   + 2*sigma_d*sigma_i**3*r[np.less(r,lo)] + 2*sigma_i**2*sigma_d**2*r[np.less(r,lo)])
+                 /(8*(sigma_i + sigma_j)**3))
+            d[np.less(r,lo)] = np.pi*P*(p2 - p1)
+        elif param == 'sigma_d':
+            p1 = (sigma_d
+                 *(6*sigma_i*sigma_d + 6*sigma_j*sigma_d + sigma_i**2 + sigma_j**2 + 8*sigma_i*sigma_j)
+                 /(6*(sigma_i + sigma_j)))
+            p2 = (sigma_i
+                 *sigma_j*(2*r[np.less(r,lo)] - sigma_i - sigma_j)*(2*sigma_d + sigma_i + sigma_j)
+                 /(8*(sigma_i + sigma_j)**2))
+            d[np.less(r,lo)] = np.pi*P*(p2 - p1)
+        else:
+            raise ValueError('The depletion parameters are P, sigma_i, sigma_j, and sigma_d.')
+
+        #clamp hi
+        hi = lo + sigma_d
+        d[np.greater(r,hi)] = 0
+
+        # evaluate in between
+        flags = np.logical_and(np.greater_equal(r,lo), np.less_equal(r,hi))
+        if param == 'P':
+            p1 = (0.5*(sigma_i + sigma_j) + sigma_d - r[flags])**2
+            p2 = r[flags]**2 + r[flags]*(sigma_i + sigma_j + 2*sigma_d) - 0.75*(sigma_i - sigma_j)**2
+            d[flags] = -(np.pi*p1*p2)/(12*r[flags])
+        elif param == 'sigma_i':
+            p1 = ((0.5*(sigma_i + sigma_j) + sigma_d - r[flags])
+                 *(r[flags]**2 + r[flags]*(sigma_i + sigma_j + 2*sigma_d) - 0.75*(sigma_i - sigma_j)**2))
+            p2 = (r[flags] + 1.5*(sigma_j - sigma_i))*(0.5*(sigma_i + sigma_j) + sigma_d - r[flags])**2
+            d[flags] = -(np.pi*P*(p1 + p2))/(12*r[flags])
+        elif param == 'sigma_j':
+            p1 = ((0.5*(sigma_i + sigma_j) + sigma_d - r[flags])
+                 *(r[flags]**2 + r[flags]*(sigma_i + sigma_j + 2*sigma_d) - 0.75*(sigma_i - sigma_j)**2))
+            p2 = (r[flags] + 1.5*(sigma_i - sigma_j))*(0.5*(sigma_i + sigma_j) + sigma_d - r[flags])**2
+            d[flags] = -(np.pi*P*(p1 + p2))/(12*r[flags])
```

Comment:
Combine these?  The derivatives have to be the same so you could use `elif (param == sigma_i) or (param == sigma_j)`